backend/eventloop/application_event_loop/application_event_runner_deployments_test.go

```diff
@@ -956,5 +966,66 @@ var _ = Describe("ApplicationEventLoop Handle deployment modified Test", func()
 			Expect(res).To(Equal(deploymentModifiedResult_NoChange),
 				"since the Namespace is being deleted, the request should not be acted upon")
 		})
+
+		It("Should verify whether ApplicationOwner row has been created, retrieved and deleted.", func() {
+			By("Create new deployment.")
```

Comment:
One more case I think we can test here:
1) Create a GitOpsDeployment/Application/ApplicationOwner like normal, like with the other tests in this file.
2) Delete the ApplicationOwner table, pointing to that Application.
3) Call 'applicationEventRunner_handleDeploymentModified()' and verify that ApplicationOwner is recreated

This is testing the update case: if we get a request for an Application, but the Application doesn't have an ApplicationOwner (for whatever reason), then we should make sure we create it. The code already does this, so it's just a matter of making sure we have a test case for this as well.